src/main/java/org/primefaces/component/fileupload/FileUpload.java

```diff
@@ -97,8 +98,13 @@ else if (newValue instanceof UploadedFiles) {
                     throw new IllegalArgumentException("Argument of type '" + newValue.getClass().getName() + "' not supported");
                 }
 
-                if (newValue instanceof UploadedFile && "advanced".equals(getMode())) {
-                    queueEvent(new FileUploadEvent(this, (UploadedFile) newValue));
+                if ("advanced".equals(getMode())) {
```

Comment:
IMO the core question is do we want to to have a chunkListener? Or are we fine with the „normal“ listener which get´s fired after the last chunk.
The code you wrote last night fired the „normal“ listener after each chunk. With my modification we fire chunkListener after each chunk and the „normal“ listener is fired after the last chunk was uploaded and we have the whole file on the server.
From my side i do not think we need the chunkListener. IMO ist kind of nice2have.

Reply:
IMO the core question is do we want to to have a chunkListener? Or are we fine with the „normal“ listener which get´s fired after the last chunk.
The code you wrote last night fired the „normal“ listener after each chunk. With my modification we fire chunkListener after each chunk and the „normal“ listener is fired after the last chunk was uploaded and we have the whole file on the server.
From my side i do not think we need the chunkListener. IMO ist kind of nice2have.

Reply:
That's actually something have been wondering as well, do we need this chunkListener...? The only use case I have in mind is whenever you want and can calculate something over chunks (as mentionned digest for example) instead of processing the whole file. Have a look at https://docs.oracle.com/javase/7/docs/api/java/security/DigestInputStream.html to have a clearer of what I'm referring to.

Right now, that's probably something we don't really need, being said, the use case I mentionned might be solved using a custom `FileUploadChunkDecoder`